ktor-server/ktor-server-core/src/io/ktor/features/ContentNegotiation.kt

```diff
@@ -53,11 +44,22 @@ class ContentNegotiation(val converters: List<ContentConverterRegistration>) {
             pipeline.sendPipeline.intercept(ApplicationSendPipeline.Render) {
                 if (subject is OutgoingContent) return@intercept
 
-                val suitableConverters = acceptedTypes().mapNotNull { (contentType, _) ->
+                val suitableConverters = call.request.acceptItems().mapNotNull { (contentType, _) ->
                     feature.converters.firstOrNull { it.contentType.match(contentType) }
                 }
 
-                val converted = suitableConverters.mapNotNull { it.converter.convertForSend(this, subject) }.firstOrNull()
+                val charset = call.request.acceptCharsetItems().mapNotNull { (charset, _) ->
+                    when {
+                        charset == "*" -> Charsets.UTF_8
+                        Charset.isSupported(charset) -> Charset.forName(charset)
+                        else -> null
+                    }
+                }.firstOrNull() ?: Charsets.UTF_8
+
+                val converted = suitableConverters.mapNotNull {
+                    val contentType = it.contentType.withCharset(charset)
+                    it.converter.convertForSend(this, contentType, subject)
```

Comment:
Same here: running every suitable converter could be too slow, need to find first, most likely we need old-school for loop here